components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -704,29 +704,33 @@ private void handleMissingSessionStateGracefully(HttpServletRequest request, Htt
             response.sendRedirect(getRedirectURL(redirectURL, request));
             return;
         }
+        String clientId = null;
+        try {
+            clientId = extractClientFromIdToken(idTokenHint);
+        } catch (ParseException e) {
+            String msg = "Error occurred while getting application information.";
+            log.error(msg, e);
+            redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
+            response.sendRedirect(getRedirectURL(redirectURL, request));
+        }
         if (!validateIdToken(idTokenHint)) {
             String msg = "ID token signature validation failed.";
-            log.error(msg);
+            log.error(msg + " Client id from id token:" + clientId);
             redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
             response.sendRedirect(getRedirectURL(redirectURL, request));
             return;
         }
         try {
-            String clientId = extractClientFromIdToken(idTokenHint);
             String callbackUrl = new OAuthAppDAO().getAppInformation(clientId).getCallbackUrl();
             if (validatePostLogoutUri(postLogoutRedirectUri, callbackUrl)) {
                 redirectURL = postLogoutRedirectUri;
             } else {
                 redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED,
                         "Post logout URI does not match with registered callback URI.");
             }
-        } catch (ParseException e) {
-            String msg = "Error occurred while getting application information.";
-            log.error(msg, e);
-            redirectURL = OIDCSessionManagementUtil.getErrorPageURL(OAuth2ErrorCodes.ACCESS_DENIED, msg);
         } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
             String msg = "Error occurred while getting application information. Client id not found.";
-            log.error(msg, e);
+            log.error(msg + " Client id from id token:" + clientId, e);
```

Comment:
Have a space after :

Reply:
Also we can add " Client id from id token:" + clientId to 'msg' string

Reply:
Add as two parts because the "msg" will be sent in the response.